src/marker.rs

```diff
@@ -830,6 +830,28 @@ impl MarkerExpression {
     }
 }
 
+impl FromStr for MarkerExpression {
+    type Err = Pep508Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let mut chars = CharIter::new(s);
+        let expression = parse_marker_key_op_value(&mut chars)?;
+        chars.eat_whitespace();
+        if let Some((pos, unexpected)) = chars.next() {
+            return Err(Pep508Error {
+                message: Pep508ErrorSource::String(format!(
+                    "Unexpected character '{}', expected end of input",
+                    unexpected
+                )),
+                start: pos,
+                len: 1,
```

Comment:
Nit. Would it make sense to compute the length of all remaining characters?

Reply:
done